kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/config/ContextTokenRestInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package io.mosip.kernel.auth.defaultadapter.config;
+
+import java.io.IOException;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import io.mosip.kernel.auth.defaultadapter.constant.AuthAdapterConstant;
+import io.mosip.kernel.core.authmanager.authadapter.model.AuthUserDetails;
+
+public class ContextTokenRestInterceptor implements ClientHttpRequestInterceptor {
+
+	@Override
+	public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution)
+			throws IOException {
+		addHeadersToRequest(request, body);
+		return execution.execute(request, body);
+	}
+	
+
+	private void addHeadersToRequest(HttpRequest httpRequest, byte[] bytes) {
+		HttpHeaders headers = httpRequest.getHeaders();
+			AuthUserDetails authUserDetails = getAuthUserDetails();
+			if (authUserDetails != null)
+				headers.set(AuthAdapterConstant.AUTH_HEADER_COOKIE,
+				AuthAdapterConstant.AUTH_COOOKIE_HEADER + authUserDetails.getToken());
+		
+	}
+
+	private AuthUserDetails getAuthUserDetails() {
+		AuthUserDetails authUserDetails = null;
+		if (SecurityContextHolder.getContext() != null && SecurityContextHolder.getContext().getAuthentication() != null
```

Comment:
From which context (spring / vertx) we have to take has to be decided here, or if we are planning implement it part another story then we have atleast have skipContextTokenReuse logic here temporarely